app/screens/home/recent_mentions/index.tsx

```diff
@@ -29,20 +102,105 @@ const RecentMentionsScreen = () => {
             transform: [{translateX: withTiming(toLeft ? -25 : 25, {duration: 150})}],
         };
-    }, [start]);
+    }, [isFocused]);
+
+    const renderItem = useCallback(({item}) => {
+        if (typeof item === 'string') {
+            if (isDateLine(item)) {
+                return (
+                    <DateSeparator
+                        date={getDateForDateLine(item)}
+                        theme={theme}
+                        timezone={isTimezoneEnabled ? currentTimezone : null}
+                    />
+                );
+            }
+            return null;
+        }
+
+        return (
+            <Mention
+                currentUser={currentUser}
+                post={item}
+            />
+        );
+    }, [currentUser]);
 
     return (
-        <SafeAreaView
-            style={{flex: 1, backgroundColor: 'brown'}}
-        >
-            <Animated.View
-                onLayout={onLayout}
-                style={[{flex: 1, justifyContent: 'center', alignItems: 'center'}, animated]}
+        <>
+            <NavigationHeader
+                isLargeTitle={isLargeTitle}
+                showBackButton={false}
+                subtitle={subtitle}
+                title={title}
+                hasSearch={false}
+                scrollValue={scrollValue}
+                forwardedRef={scrollRef}
+            />
+            <SafeAreaView
+                style={styles.flex}
+                edges={EDGES}
             >
-                <Text style={{fontSize: 20, color: '#fff'}}>{'Recent Mentions Screen'}</Text>
-            </Animated.View>
-        </SafeAreaView>
+                <Animated.View style={[styles.flex, animated]}>
+                    {loading && !mentions.length ? (
+                        <View style={styles.loading}>
+                            <ActivityIndicator
+                                color={theme.sidebarBg}
+                                size='large'
+                            />
+                        </View>
+                    ) : (
+                        <AnimatedFlatList
+                            ref={scrollRef}
+                            contentContainerStyle={paddingTop}
+                            data={posts}
+                            scrollToOverflowEnabled={true}
+                            showsVerticalScrollIndicator={false}
+                            progressViewOffset={scrollPaddingTop}
+                            scrollEventThrottle={16}
+                            indicatorStyle='black'
+                            onScroll={onScroll}
+                            onRefresh={handleRefresh}
+                            refreshing={refreshing}
+                            renderItem={renderItem}
+                        />
+                    )}
+                </Animated.View>
+            </SafeAreaView>
+        </>
     );
 };
 
-export default RecentMentionsScreen;
+const enhance = withObservables([], ({database}: WithDatabaseArgs) => {
```

Comment:
we want to keep this in the index file (it can be renamed to index.ts) and move the component to a `recent_mentions.tsx` file